platform/mellanox/mlnx-platform-api/sonic_platform/component.py

```diff
@@ -684,11 +684,11 @@ def get_available_firmware_version(self, image_path):
     def get_firmware_update_notification(self, image_path):
         return "Immediate cold reboot is required to complete {} firmware update".format(self.name)
 
-    def install_firmware(self, image_path):
-        return self.__install_firmware(image_path)
+    def install_firmware(self, image_path, allow_reboot=True):
+        return self.__install_firmware(image_path, allow_reboot)
 
     def update_firmware(self, image_path):
-        self.__install_firmware(image_path)
+        self.__install_firmware(image_path, reboot=True)
```

Comment:
@alexrallen why need to explicitly pass this flag?

Reply:
We don't